### DISCUSSION TITLE
Need help publishing to github pages from a GitHub Actions workflow

### DISCUSSION THREAD

**Discussion by jamiet-msm** [community/community#41627] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I am trying follow the instructions at [Creating a custom GitHub Actions workflow to publish your site](https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site) to publish artifacts from a GitHub Actions workflow to GitHub pages. I think I've followed the correct steps but I cannot see the content where I am expecting to see it.

The artifact I'm trying to publish is a simple .svg file that I want to refer to in my repo's README.

The most recent run of my workflow job is at: https://github.com/jamiekt/jstark/actions/runs/3683869250/jobs/6233204182. As per [the instructions provided for the upload-pages-artifact action](https://github.com/actions/upload-pages-artifact)

> A Pages artifact must:
>
> * Be called github-pages
> * Be a single gzip archive containing a single tar file

upload-pages-artifacts/action.yml also states that the path containing the assets defaults to **_site**:

```
    description: "Path of the directory containing the static assets."
    required: true
    default: "_site/"
```
[action.yml#L10-L13](https://github.com/actions/upload-pages-artifact/blob/c8641e80048cf33d329c39df3bcc53196178335a/action.yml#L10-L13)

Hence, in my workflow job I am creating  **_site/github-pages.tar.gz**:

```shell
> tar -ztvf _site/github-pages.tar.gz
drwxr-xr-x runner/docker     0 2022-12-13 09:21 artifacts-for-pages/
-rw-r--r-- runner/docker   901 2022-12-13 09:21 artifacts-for-pages/coverage.svg
```
so I think I have satisfied all the criteria.

It then gets uploaded and deployed successfully apparently:

> {"page_url":"https://jamiekt.github.io/jstark/","status_url":"https://api.github.com/repos/jamiekt/jstark/pages/deployment/status/ccfcfe27fc5a568b34a0ee3b88eb05ccd731c120","preview_url":""}

So I'm hoping to see some content at one of these URLs:

* https://jamiekt.github.io/jstark/coverage.svg
* https://jamiekt.github.io/jstark/artifacts-for-pages/coverage.svg

but they're returning 404s.

I don't know what I've done wrong here, can anyone advise? Why do I not see any content on my GitHub pages site?

**Comment by yoannchaudet** [comment]
Hello here,

All the `tar` handling is already done for you in [`actions/upload-pages-artifact`](https://github.com/actions/upload-pages-artifact). 

The problem here is that you are trying to do what the Action does for you so you end up with another `tar` file being served on your site: https://jamiekt.github.io/jstark/github-pages.tar.gz.

Here is a two-jobs example you can model your workflow on: https://github.com/actions/starter-workflows/blob/main/pages/hugo.yml.

Unless you need a dedicated "badge" artifact I suggest uploading the Pages artifact directly in your `run` job. Something like that ~:

```
- run: mkdir _site && cp coverage.svg _site/.
- name: Upload GitHub Pages artifact
  uses: actions/upload-pages-artifact@v1
```

In your `deploy-to-github-pages` job you only need:

```
- name: Deploy to GitHub Pages
  id: deployment
  uses: actions/deploy-pages@v1
```

**Reply by jamiet-msm** [reply]
Fantastic, that worked (here is the proof: https://jamiekt.github.io/jstark/coverage.svg), thank you very much.

**Comment by ramez0909** [comment]
hello 
i have made all steps but it's not working and Containo saying error 404
